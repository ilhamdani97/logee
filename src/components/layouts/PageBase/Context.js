import { createContext } from 'react';


export const PageBaseContext = createContext(null);
