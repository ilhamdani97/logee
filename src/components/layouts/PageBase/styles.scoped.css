.main {
  padding-top: 48px;
}

.separator {
  width: 100%;
  height: 0.0625rem;
  margin: 0.75rem 0;
  background-color: var(--grey-40);
}

.noPadding {
  padding: 0;
}

@media (min-width: 768px) {
  .main {
    display: block;
    margin-left: 16rem;
    padding-top: 3.7rem;
    padding-bottom: 2.5rem;
    padding-right: 2.5rem;
    padding-left: 2.5rem;
  }
}
