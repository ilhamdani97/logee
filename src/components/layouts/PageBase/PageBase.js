import React, { useState , useEffect } from 'react';
import { useLocation } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';
import DashboardHeader from '@components/layouts/DashboardHeader';
import { PageBaseContext } from './Context';
import { routes } from '@configs/routes';
import DashboardMenu from '@layouts/DashboardMenu';

export default function PageBase({ children }) {
  const { pathname } = useLocation();
  const [showMenu,setShowMenu] = useState(false);
  const [router,setRouter] = useState('');

  const noAuthRoutes = [
    `${routes.LOGIN()}`];
  const pageWithoutMenu = noAuthRoutes.some(r => pathname.match(r));

  useEffect(() => {
    pathname === '/' && setRouter(null);
  }, [pathname]);


  if (pageWithoutMenu) {
    return children;
  }


  return (
    <PageBaseContext.Provider value={{ showMenu,setShowMenu,router, setRouter }}>
      <DashboardHeader />
      <DashboardMenu />
      <main className={styles.main}>
        {children}
      </main>
    </PageBaseContext.Provider>
  );
}

PageBase.defaultProps = {
  children: null,
};

PageBase.propTypes = {
  children: PropTypes.node,
};
