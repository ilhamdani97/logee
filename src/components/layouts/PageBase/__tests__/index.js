import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import PageBase from '../PageBase';


describe('src/pages/PageBase', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<PageBase />);
    expect(tree).toMatchSnapshot();
  });
});
