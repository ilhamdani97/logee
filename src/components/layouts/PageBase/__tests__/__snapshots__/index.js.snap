// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/PageBase render 1`] = `
<Provider
  value={
    Object {
      "router": "",
      "setRouter": [MockFunction] {
        "calls": Array [
          Array [
            null,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "setShowMenu": [MockFunction],
      "showMenu": false,
    }
  }
>
  <DashboardHeader />
  <DashboardMenu />
  <main />
</Provider>
`;
